python/cudf/cudf/_lib/groupby.pyx

```diff
@@ -62,19 +66,45 @@ _DECIMAL_AGGS = {"COUNT", "SUM", "ARGMIN", "ARGMAX", "MIN", "MAX", "NUNIQUE",
 # workaround for https://github.com/cython/cython/issues/3885
 ctypedef const scalar constscalar
 
+
+cdef _agg_result_from_columns(
+    vector[libcudf_groupby.aggregation_result]& c_result_columns,
+    set column_orders,
```

Comment:
I would rename this variable (both here and in all of the code calling `_agg_result_from_columns`) to something like `columns_included`. It no longer contains any ordering, it is just a list of indexes. The order is controlled by the iteration through the ordered list of columns returned from libcudf.